src/Components/Components/test/Routing/QueryParameterValueSupplierTest.cs

```diff
@@ -1,98 +1,53 @@
-// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
The git diff is pretty useless in this file, so I can't really review the change without completely reverse-engineering it, which would be more work than implementing the change in the first place.

However if you think no behavioral changes have been introduced, and there's nothing breaking for us to be concerned about, that sounds good to me.

Reply:
Most of the tests were refactored to account for the fact that `QueryParameterValueSupplier` no longer has any logic determining which component properties should have values supplied (because that's now decided by the existing cascading parameter mechanism). The diff is so crazy because the test previously verified parameter setting logic, but now it's just validating the query parameter parsing logic. The cases that got removed from this test file should be covered by the existing E2E and cascading value tests.

However, the one thing I _know_ has changed is that we no longer throw an exception if two component parameters read the same query value. I think this is probably a fine change because:
1. It's not breaking (a case works that didn't used to)
2. It's consistent with how other cascading parameter types behave
3. The reason we didn't previously make it work before (from what I can see) was that `QueryParameterValueSupplier` internally kept a 1:1 mapping between query parameter names and component properties. So this exception was added as an alternative to silently invalidating the 1:1 mapping assumption.

If you disagree and think we should still throw an exception for this case, I'm happy to address this in a follow-up.

Reply:
That's really helpful. Thanks for the analysis and description.

> However, the one thing I know has changed is that we no longer throw an exception if two component parameters read the same query value

Sounds absolutely fine to me. Making an extra case work, when it's something sensible and reasonable, is an extra bonus. I don't know of any reason we'd want to stop people from doing this.